### PR TITLE
Create tutorial about basic templated one-shot tests.

### PR DESCRIPTION
**Pull request by will-cromar**
I had to do some hackery to make it so the monitor container generates for our tests but not the tutorial examples. Ideally, this should be an optional feature, and you should be able to easily specify the right image for your project. I'll file a larger bug about separating the reusable library components of our templates from the configuration that we use in our project.

### CODE REVIEW

File: doc/tutorials/test-templates-basic.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
+# Creating Tests with JSonnet Templates
+
+## Prerequisites
+
+1. A clone of this repository.
+1. Build dependencies from [developing.md](/doc/developing.md).
```
Comment (zcain117) [inline doc/tutorials/test-templates-basic.md:6]: another prereq is a GCP project. Maybe link to some page on setting up a GCP project

File: doc/tutorials/test-templates-basic.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+1. A GKE cluster with GPUs and (optionally) Cloud TPUs. Accelerator availability depends on GCE zone. All of the accelerators used in this tutorial are available in `us-central1-b`.
+  - Example command:
+    ```
+      gcloud beta container clusters create tutorial-cluster \
```
Comment (zcain117) [inline doc/tutorials/test-templates-basic.md:10]: I think we should either add an explicit `--project` flag or make a very visible note that this command will use the current default project in the user's shell

File: doc/tutorials/test-templates-basic.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    ```
+1. A GCS bucket.
+  - To create a new GCS bucket, run `gsutil mb -c standard -l us-central1 gs://your-bucket-name`
+
+## Creating a simple test without accelerators
+
+We'll start by creating a simple test that runs the example MNIST model from the TensorFlow model garden on CPUs. First, build the TensorFlow test image from this repository:
+
+```bash
+gcloud builds submit --config images/tensorflow/cloudbuild.yaml --project=xl-ml-test --substitutions _VERSION=r2.2,_BASE_IMAGE_VERSION=2.2.0,_MODEL_GARDEN_BRANCH=r2.2.0
```
Comment (zcain117) [inline doc/tutorials/test-templates-basic.md:30]: I'd make this an env variable. People like to copy-paste without reading first :)

File: doc/tutorials/test-templates-basic.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+```
+
+This new image will be available at `gcr.io/your-project/tensorflow:r2.2.
```
Comment (zcain117) [inline doc/tutorials/test-templates-basic.md:33]: Link to images page in cloud console so user can double check: https://console.cloud.google.com/gcr

File: doc/tutorials/test-templates-basic.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+You can find the job running on your [GKE workloads](https://console.cloud.google.com/kubernetes/workload) page.
+
+## Storing datasets on GCS
+
+The above example downloads the MNIST dataset every time it runs. That's workable for a small dataset like MNIST, but that's not practical for large datasets. The easiest way to store the dataset is on GCS.
+
+For this example, we can download the MNIST dataset with TensorFlow Datasets (TFDS). Either install the `tensorflow-datasets` pip package or run a Docker image that already has it installed:
+
+```bash
+kubectl run -it --rm --image gcr.io/your-project/tensorflow:r2.2 --env=GCS_BUCKET=$GCS_BUCKET $USER bash
```
Comment (zcain117) [inline doc/tutorials/test-templates-basic.md:162]: this could use the project env var too

File: doc/tutorials/test-templates-basic.md (hunk 1 above)
Comment (will-cromar) [reply doc/tutorials/test-templates-basic.md:7]: Done.

File: doc/tutorials/test-templates-basic.md (hunk 2 above)
Comment (will-cromar) [reply doc/tutorials/test-templates-basic.md:11]: Done.

File: doc/tutorials/test-templates-basic.md (hunk 3 above)
Comment (will-cromar) [reply doc/tutorials/test-templates-basic.md]: Good idea. Done.

File: doc/tutorials/test-templates-basic.md (hunk 4 above)
Comment (will-cromar) [reply doc/tutorials/test-templates-basic.md]: Done.

File: doc/tutorials/test-templates-basic.md (hunk 5 above)
Comment (will-cromar) [reply doc/tutorials/test-templates-basic.md]: Done

**Pull request merged**